source/features/unfinished-comments.tsx

```diff
@@ -1,26 +1,40 @@
 import select from 'select-dom';
+import delegate from 'delegate-it';
 import * as pageDetect from 'github-url-detection';
 
 import features from '.';
 
 let documentTitle: string | undefined;
 
 function hasDraftComments(): boolean {
-	return select.all('textarea').some(textarea => textarea.value.length > 0 && textarea.offsetWidth > 0);
+	// `[disabled]` excludes the PR description field that `wait-for-build` disables while it waits
+	return select.all<HTMLTextAreaElement>('textarea:not([disabled])').some(textarea =>
+		textarea.value !== textarea.textContent && // Exclude comments being edited but not yet changed (and empty comment fields)
+		textarea.offsetWidth > 0 && // Exclude invisible fields
+		!textarea.closest('.rgh-is-sending-comment') // Exclude forms being submitted
```

Comment:
I think this would cover:

- regular comments
- review comments
- PR merges

```suggestion
		select.exists('.btn-primary[disabled]', textarea.form!);
```

This reads as: if the form of the current textarea (which is filled, changed, and visible) contains a disabled "submit" button, then the textarea is being sent.


I'm not sure if the selector needs to be tweaked.

Reply:
Tested all three cases and they work great! Although the feature doesn't work when creating a new issue/PR — but I don't think it was the goal in the first place?

Reply:
I think that's just due to `pageDetect.hasComments`. Perhaps the correct one is `hasRichTextEditor`